paasta_tools/setup_marathon_job.py

```diff
@@ -294,49 +304,74 @@ def log_bounce_action(line, level='debug'):
         )
 
 
-def get_old_happy_unhappy_draining_tasks_for_app(app, drain_method, service, nerve_ns, bounce_health_params,
-                                                 system_paasta_config):
+def get_tasks_by_state_for_app(app, drain_method, service, nerve_ns, bounce_health_params,
+                               system_paasta_config):
     tasks_by_state = {
         'happy': set(),
         'unhappy': set(),
         'draining': set(),
+        'at_risk': set(),
     }
 
     happy_tasks = bounce_lib.get_happy_tasks(app, service, nerve_ns, system_paasta_config, **bounce_health_params)
+    draining_hosts = get_draining_hosts()
     for task in app.tasks:
         if drain_method.is_draining(task):
             state = 'draining'
         elif task in happy_tasks:
-            state = 'happy'
+            if task.host in draining_hosts:
+                state = 'at_risk'
+            else:
+                state = 'happy'
         else:
             state = 'unhappy'
         tasks_by_state[state].add(task)
 
     return tasks_by_state
 
 
-def get_old_happy_unhappy_draining_tasks(other_apps, drain_method, service, nerve_ns, bounce_health_params,
-                                         system_paasta_config):
-    """Split tasks from old apps into 3 categories:
+def get_tasks_by_state(other_apps, drain_method, service, nerve_ns, bounce_health_params,
+                       system_paasta_config):
+    """Split tasks from old apps into 4 categories:
       - live (not draining) and happy (according to get_happy_tasks)
       - live (not draining) and unhappy
       - draining
+      - at-risk (running on a host marked draining in Mesos in preparation for maintenance)
     """
 
     old_app_live_happy_tasks = {}
     old_app_live_unhappy_tasks = {}
     old_app_draining_tasks = {}
+    old_app_at_risk_tasks = {}
 
     for app in other_apps:
 
-        tasks_by_state = get_old_happy_unhappy_draining_tasks_for_app(
+        tasks_by_state = get_tasks_by_state_for_app(
             app, drain_method, service, nerve_ns, bounce_health_params, system_paasta_config)
 
         old_app_live_happy_tasks[app.id] = tasks_by_state['happy']
         old_app_live_unhappy_tasks[app.id] = tasks_by_state['unhappy']
         old_app_draining_tasks[app.id] = tasks_by_state['draining']
+        old_app_at_risk_tasks[app.id] = tasks_by_state['at_risk']
+
+    return old_app_live_happy_tasks, old_app_live_unhappy_tasks, old_app_draining_tasks, old_app_at_risk_tasks
 
-    return old_app_live_happy_tasks, old_app_live_unhappy_tasks, old_app_draining_tasks
+
+def get_num_at_risk_tasks(app):
```

Comment:
Yep, still working on this. It is a bit annoying due to having to mock out a bunch of the paasta_maintenance stuff (due to the dependency on mesos.cli's config). This is what led me to discover that we can't safely run our itests individually (due to not properly setting up and cleaning up), and I got a bit distracted trying to sort that out. But thanks for the reminder :)